using-mobile-extensions/adobe-edge-consent/README.md

```diff
@@ -0,0 +1,165 @@
+
+   ```swift
+   use_frameworks!
+   target 'YourTargetApp' do
+       pod 'AEPCore'
+       pod 'AEPLifecycle'
+       pod 'AEPSignal'
+       pod 'AEPEdge'
+       pod 'AEPEdgeIdentity'
+       pod 'AEPEdgeConsent'
+   end
+   ```
+
+2. Import the Mobile Core and Edge libraries:
+
+### Swift
+
+```swift
+// AppDelegate.swift
+import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPEdgeConsent
+import AEPLifecycle
+import AEPSignal
+```
+
+### Objective-C
+
+```text
+// AppDelegate.h
+@import AEPCore;
+@import AEPEdge;
+@import AEPEdgeIdentity;
+@import AEPEdgeConsent;
+@import AEPLifecycle;
+@import AEPSignal;
+```
+
+{% endtab %}
+{% endtabs %}
+
+### Register Edge with Mobile Core
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+```java
+public class MobileApp extends Application {
+
+    @Override
+    public void onCreate() {
+      super.onCreate();
+      MobileCore.setApplication(this);
+      MobileCore.configureWithAppID("yourAppId");
+
+      try {
+        Edge.registerExtension();
+        Consent.registerExtension(); // register Consent
+        com.adobe.marketing.mobile.edge.identity.Identity.registerExtension();
+        Lifecycle.registerExtension();
+        Signal.registerExtension();
+        MobileCore.start(new AdobeCallback() {
+          @Override
+          public void call(final Object o) {
+            // processing after start
+          }});
+      } catch (Exception e) {
+        //Log the exception
+      }
+    }
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+### Swift
+
+```swift
+// AppDelegate.swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    MobileCore.registerExtensions([Lifecycle.self, Identity.self, Signal.self, Edge.self, Consent.self], {
+    MobileCore.configureWith(appId: "yourAppId")
+  })
+  ...
+}
+```
+
+### Objective-C
+
+```text
+// AppDelegate.m
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    [AEPMobileCore registerExtensions:@[AEPMobileEdgeIdentity.class, AEPMobileLifecycle.class, AEPMobileSignal.class, AEPMobileEdge.class, AEPMobileEdgeConsent.class] completion:^{
+    ...
+  }];
+  [AEPMobileCore configureWithAppId: @"yourAppId"];
+  ...
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Configuration keys
+
+To update the SDK configuration programmatically, use the following information to change the Edge Consent configuration values.
+
+| Key             | Required | Description                                                  | Data Type |
+| :-------------- | :------- | :----------------------------------------------------------- | :-------- |
+| consent.default | No       | Consents in XDM format. [More info.](https://github.com/adobe/xdm/blob/fc0773107f29928e1dc4753f8f055836083ea53f/docs/reference/mixins/profile/profile-consents.schema.md) | String    |
+
```

Comment:
what do u think about providing an example? May be call the API with default collect consent "y"

Reply:
> Next PR will follow with API spec.

This README is just how to get setup, not use the APIs.